File: historian-core/src/main/java/net/yslibrary/historian/internal/LogWritingTask.java
package net.yslibrary.historian.internal;

import net.yslibrary.historian.Historian;

/**
 * Runnable implementation writing logs and executing callbacks
 */

public class LogWritingTask implements Runnable {

  private final Historian.Callbacks callbacks;
  private final LogWriterDB logWriterDB;
  private final LogEntity log;

  public LogWritingTask(Historian.Callbacks callbacks,
                        LogWriterDB logWriterDB,
                        LogEntity log) {
    this.callbacks = callbacks;
    this.logWriterDB = logWriterDB;
    this.log = log;
  }

  @Override
  public void run() {

    try {
      logWriterDB.log(log);

      callbacks.onSuccess();
    } catch (final Throwable throwable) {
      callbacks.onFailure(throwable);
    }
  }
}

File: historian-core/src/main/java/net/yslibrary/catlog/internal/LogWritingTask.kt
package net.yslibrary.catlog.internal

import net.yslibrary.catlog.CatLog

/**
 * Runnable implementation writing logs and executing callbacks
 */

class LogWritingTask(private val callbacks: CatLog.Callbacks,
                     private val logWriter: LogWriter,
                     private val log: LogEntity) : Runnable {

    override fun run() {
        try {
            logWriter.log(log)
            callbacks.onSuccess()
        } catch (throwable: Throwable) {
            callbacks.onFailure(throwable)
        }

    }
}
